src/zcash/JoinSplit.cpp

```diff
@@ -141,10 +142,26 @@ class JoinSplitCircuit : public JoinSplit<NumInputs, NumOutputs> {
         }
 
         r1cs_ppzksnark_proof<ppzksnark_ppT> r1cs_proof;
-        std::stringstream ss;
-        std::string proof_str(proof.begin(), proof.end());
-        ss.str(proof_str);
-        ss >> r1cs_proof;
+        {
+            LOCK(cs_PointCompression);
```

Comment:
This is not threadsafe despite the lock. For it to be threadsafe, all other users of the serialization that rely on ``g[12]_enable_point_compression`` not changing, would have to also take the lock.